src/jit/morph.cpp

```diff
@@ -4451,15 +4451,16 @@ GenTreeCall* Compiler::fgMorphArgs(GenTreeCall* call)
     }
 #endif // FEATURE_FIXED_OUT_ARGS
 
-    /* Update the 'side effect' flags value for the call */
-
-    call->gtFlags |= (flagsSummary & GTF_ALL_EFFECT);
-
-    if (!call->OperMayThrow(this) && ((flagsSummary & GTF_EXCEPT) == 0))
+    // Clear the ASG and EXCEPT (if possible) flags on the call node
+    call->gtFlags &= ~GTF_ASG;
+    if (!call->OperMayThrow(this))
     {
         call->gtFlags &= ~GTF_EXCEPT;
     }
 
+    // Union in the side effect flags from the call's operands
+    call->gtFlags |= flagsSummary & GTF_ALL_EFFECT;
```

Comment:
CC:  (flagsSummary & GTF_ALL_EFFECT)

Reply:
IIUC the parens are only necessary on subexpressions (e.g. `call->gtFlags = call->gtFlags | (flagsSummary & GTF_ALL_EFFECT)`

Reply:
It is a subexpression of |= here and it looks like such subexpressions are parenthesized in most places in the codebase but I'll leave this up to you.